### ISSUE TITLE
Figure out meaning of level chunk L51 (Loop Table)

### ISSUE THREAD

**Issue by dertseha** [inkyblackness/ss-specs#6]
Level chunk L51 contains 0x03C0 bytes for each level and appears to contain up to 64 entries of 15 bytes length. Try to figure out what it is for.

**Comment by dertseha** [comment]
Theory: Is this some level object specific state data?

Assumptions:

```
0000  int16   entry type (seen 0x0080 and 0x0020)
0002  int16   index to master object table (found values appear to be within limits)
0004  byte?   frame/state? Two savegames with ~2 seconds difference compared have this value increased.
```

**Comment by dertseha** [comment]
I did some tests with this table. That second int16 is indeed the index into the master table. Most of the entries that I looked at refer to screens. There was also one door there.

That extra byte - at least for the screen that I played with - controls the loop type (forward, backward, ...) I didn't verify it - on level 1 there is that one door that constantly opens and closes, perhaps this one is referred to the one door entry.
I coined it to be a LoopTable for now.

More tests will make it clearer what the values mean.

The hacker tool should also be extended to read this table, as well as extend some query.
I wonder if one of the master object tables has an index into this table; Otherwise it would be a one-way only.

**Comment by jramstedt** [comment]
I did some research and this is the current guess

```
0000 int16   Index to master object table
0002 byte    Loop wrap mode (seems to be modulo 8)
0003 byte[10] Unknown
0013 int16   Frame time
```

Loop animation FPS is calculated dividing 256 with frame time.
fps = 256 / frame time

**Comment by jramstedt** [comment]
The opening and closing door in medical is controlled by repeat action trigger. That trigger is activated by "enter trigger" when you walk to the door from the gamma corridor. That trigger is somewhere between "mini-grove" and the door. If you go to the door through the maintenance section (elevator with wire puzzle near the start), door will not open and close (because the repeating trigger isn't activated).

**Comment by dertseha** [comment]
Revisiting this table now I found out it only works for modifying save-games. The authoritative information about looped items are the level objects. I realized this by creating a screen object with the intention of altering the table afterwards. In my test, the screen already properly looped (as per level object properties), and this table was initialized, referring to the screen object. Cross-testing this I filled this table in archive.dat by hand and started the game - the screen still updated according to the default timing value and not what I had set. The save-game even had my values overwritten.

Also, thank you @jramstedt for the proper layout - I have no idea why I misaligned it in the first place :)

For now I will consider this table as a by-product / runtime scratch-pad of the engine.
